plugins/plugin-java/che-plugin-java-ext-jdt/che-jdt-ext-machine/src/main/java/org/eclipse/che/jdt/rest/CompilerSetupService.java

```diff
@@ -87,4 +115,31 @@ public void setParameters(@QueryParam("projectpath") String projectPath, @NotNul
         //noinspection unchecked
         return options.getMap();
     }
+
+    private void updateCreateSettingsFile(Map<String, String> changedParameters) throws IOException {
+        File settingsFile = Paths.get(settingsDir, preferencesFileName).toFile();
+        createParentDirs(settingsFile);
+
+        String content = "";
```

Comment:
note that JDK compiler uses StringBuilder in bytecode when using + in string concat